src/Microsoft.DotNet.TestFramework/TestAssets.cs

```diff
@@ -15,14 +15,36 @@ public class TestAssets
     {
         private DirectoryInfo _root;
 
-        public TestAssets(DirectoryInfo assetsRoot)
+        private FileInfo _dotnetCsprojExe;
+
+        private FileInfo _dotnetProjectJsonExe;
+
+        private const string ProjectJsonSearchPattern = "project.json";
+
+        private const string CsprojSearchPattern = "*.csproj";
+
+        public TestAssets(DirectoryInfo assetsRoot, FileInfo dotnetCsprojExe, FileInfo dotnetProjectJsonExe)
```

Comment:
Particularly for public classes, it's worth adding guard clauses to throw an explicit `ArgumentNullException` rather than a relatively less-helpful `NullReferenceException`.